src/modules/commander/Commander.cpp

```diff
@@ -1547,9 +1548,12 @@ Commander::run()
 			_was_falling = _land_detector.freefall;
 		}
 
+		if (_controller_status_sub.updated()) {
```

Comment:
Move this within the armed.armed conditional and keep a local launch_detection_running bool. The idea is to minimize the work you do (only update when armed) and minimize memory usage (only store the minimum).

Reply:
Suggested changes added. I'm not completely sure I put the the launch_detection_running bool in the proper place.